plugins/controls.ts

```diff
@@ -0,0 +1,129 @@
+import { watchDebounced } from '@vueuse/core'
+import { type InjectionKey } from 'vue'
+export const ControlsSymbol: InjectionKey<ReturnType<typeof controlsService>> = Symbol('controls')
+
+import Item from '~/models/Item'
+interface IFilterConfig {
+  sortBy: string
+  sortDirection?: 'asc' | 'desc'
+  perPage: number
+}
+
+interface Response {
+  data: Item[]
+  total: number
+  current_page: number
+  last_page: number
+}
+
+const controlsService = async (
+  options: IFilterConfig = {
+    sortBy: 'updated_at',
+    sortDirection: 'asc',
+    perPage: 12,
+  }
+) => {
+  const nuxtConfig = useRuntimeConfig()
+  const route = useRoute()
+  const router = useRouter()
+  const filters = reactive<Record<string, any>>({})
+  const routeParams = reactive<Record<string, string>>({})
+  const aggregations = reactive<Record<string, unknown>>({
+    size: 1000,
+  })
+
+  const page = ref(route.query.page ? Number(route.query.page) : 1)
+  const sortBy = ref(route.query.sortBy || options.sortBy)
+  const sortDirection = ref(route.query.sortDirection || options.sortDirection)
+
+  const items = ref<any[]>([])
+
+  const aggregationQuery = computed(() => ({ ...filtersQuery.value, ...aggregations }))
+
+  const filtersQuery = computed(() => {
+    const query = {
+      page: page.value,
+      size: options.perPage,
+      [`sort[${sortBy.value}]`]: sortDirection.value,
+    }
+
+    return {
+      ...query,
+      ...filters,
+    }
+  })
+
+  const [itemsDataFetch, aggDataFetch] = await Promise.all([
+    useFetch<Response>('api/v1/items', {
+      baseURL: nuxtConfig.public.APP_URL,
+      query: filtersQuery,
+      watch: [filtersQuery],
+      transform: (response) => ({
+        ...response,
+        data: response.data.map((data) => new Item(data)),
+      }),
+      immediate: false,
+    }),
```

Comment:
It's not necessary, but this would de-couple us from both the config and the `Item` class

```suggestion
    useFetchItems({
      query: filtersQuery,
      watch: [filtersQuery],
      immediate: false,
    }),
```